datasets/qed/qed.py

```diff
@@ -0,0 +1,137 @@
+    author={Matthew Lamm and Jennimaria Palomaki and Chris Alberti and Daniel Andor and Eunsol Choi and Livio Baldini Soares and Michael Collins},
+    year={2020},
+    eprint={2009.06354},
+    archivePrefix={arXiv},
+    primaryClass={cs.CL}
+}
+"""
+
+_DESCRIPTION = """\
+QED, is a linguistically informed, extensible framework for explanations in question answering. \
+A QED explanation specifies the relationship between a question and answer according to formal semantic notions \
+such as referential equality, sentencehood, and entailment. It is an expertannotated dataset of QED explanations \
+built upon a subset of the Google Natural Questions dataset.
+"""
+
+_HOMEPAGE = "https://github.com/google-research-datasets/QED"
+
+_BASE_URL = "https://raw.githubusercontent.com/google-research-datasets/QED/master/"
+_URLS = {
+    "train": _BASE_URL + "qed-train.jsonlines",
+    "dev": _BASE_URL + "qed-dev.jsonlines",
+}
+
+
+class Qed(datasets.GeneratorBasedBuilder):
+    """QED: A Dataset for Explanations in Question Answering"""
+
+    VERSION = datasets.Version("1.0.0")
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="qed", version=datasets.Version("1.0.0")),
+    ]
+
+    def _info(self):
+        span_features = {
+            "start": datasets.Value("int32"),
+            "end": datasets.Value("int32"),
+            "string": datasets.Value("string"),
+        }
+        reference_features = {
+            "start": datasets.Value("int32"),
+            "end": datasets.Value("int32"),
+            "bridge": datasets.Value("bool_"),
+            "string": datasets.Value("string"),
+        }
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "example_id": datasets.Value("int64"),
+                    "title_text": datasets.Value("string"),
+                    "url": datasets.Value("string"),
+                    "question": datasets.Value("string"),
+                    "paragraph_text": datasets.Value("string"),
+                    "sentence_starts": datasets.Sequence(datasets.Value("int32")),
+                    "original_nq_answers": [[span_features]],
+                    "annotation": {
+                        "referential_equalities": [
+                            {
+                                "question_reference": span_features,
+                                "sentence_reference": reference_features,
+                            }
+                        ],
+                        "answer": [
+                            {
+                                "sentence_reference": reference_features,
+                                "paragraph_reference": span_features,
+                            }
+                        ],
+                        "explanation_type": datasets.Value("string"),
+                        "selected_sentence": span_features,
+                    },
```

Comment:
could we simplify this using less nested structures ?

for example
```
"sentence_answers": datasets.Sequence(span_features),
"paragraph_answers": datasets.Sequence(span_features),
"sentence_referential_equalities": datasets.Sequence(reference_features),
"paragraph_referential_equalities": datasets.Sequence(reference_features),
"explanation_type": datasets.Value("string"),
"selected_sentence": span_features,
```
let me know what you think

Reply:
I think it would be better to stick with author's conventions. The nested structure seems conceptually easier to understand.  Also pinging @yjernite  about this.

Reply:
We can keep the original structure to stay coherent